hub/core/storage/memory.py

```diff
@@ -1,19 +1,86 @@
-from hub.core.storage.mapped_provider import MappedProvider
-import fsspec  # type: ignore
+from hub.core.storage.provider import StorageProvider
 
 
-class MemoryProvider(MappedProvider):
+class MemoryProvider(StorageProvider):
     """Provider class for using the memory."""
 
     def __init__(self, root):
-        """Initializes the MemoryProvider.
+        self.mapper = {}
+
+    def __getitem__(
+        self,
+        path: str,
+    ):
+        """Gets the object present at the path within the given byte range.
+
+        Example:
+            memory_provider = MemoryProvider("xyz")
+            my_data = memory_provider["abc.txt"]
+
+        Args:
+            path (str): The path relative to the root of the provider.
+
+        Returns:
+            bytes: The bytes of the object present at the path.
+
+        Raises:
+            KeyError: If an object is not found at the path.
+        """
+        return self.mapper[path]
```

Comment:
if mapper is just a dict, then this is definitely not thread safe right? i thought we definitely needed that (& fsspec supports this)

Reply:
Yup it's not. I can make it threadsafe later when we need it, but fsspec's implementation was 50x slower